nvflare/app_opt/confidential_computing/cc_manager.py

```diff
@@ -0,0 +1,234 @@
+            return (
+                f"bad PEER_CTX_CC_TOKEN prop in peer context: must be a dict but got {type(participants_to_validate)}"
+            )
+
+        if not participants_to_validate:
+            return ""
+
+        return self._validate_participants_tokens(participants_to_validate)
+
+    def _server_to_check_client_token(self, fl_ctx: FLContext) -> str:
+        participants = fl_ctx.get_prop(FLContextKey.JOB_PARTICIPANTS)
+        if not participants:
+            return f"missing '{FLContextKey.JOB_PARTICIPANTS}' prop in fl_ctx"
+
+        if not isinstance(participants, list):
+            return f"bad value for {FLContextKey.JOB_PARTICIPANTS} in fl_ctx: expect list bot got {type(participants)}"
+
+        participant_tokens = {self.site_name: self.my_token}
+        for p in participants:
+            assert isinstance(p, str)
+            if p == self.site_name:
+                continue
+            if p not in self.participant_cc_info:
+                return f"no token available for participant {p}"
+            participant_tokens[p] = self.participant_cc_info[p][CC_TOKEN]
+
+        err = self._validate_participants_tokens(participant_tokens)
+        if err:
+            return err
+
+        for p in participant_tokens:
+            self.participant_cc_info[p][CC_TOKEN_VALIDATED] = True
+        fl_ctx.set_prop(key=PEER_CTX_CC_TOKEN, value=participant_tokens, sticky=True, private=False)
+        self.logger.debug(f"{self.site_name=} set PEER_CTX_CC_TOKEN with {participant_tokens=}")
+        return ""
+
+    def _validate_participants_tokens(self, participants) -> str:
+        self.logger.debug(f"Validating participant tokens {participants=}")
+        result = self.helper.validate_participants(participants)
+        assert isinstance(result, dict)
+        for p in result:
+            self.participant_cc_info[p] = {CC_TOKEN: participants[p], CC_TOKEN_VALIDATED: True}
+        invalid_participant_list = [k for k, v in self.participant_cc_info.items() if v[CC_TOKEN_VALIDATED] is False]
+        if invalid_participant_list:
+            invalid_participant_string = ",".join(invalid_participant_list)
+            self.logger.debug(f"{invalid_participant_list=}")
+            return f"Participant {invalid_participant_string} not meeting CC requirements"
+        else:
+            return ""
+
+    def _not_authorize_job(self, reason: str, fl_ctx: FLContext):
+        job_id = fl_ctx.get_prop(FLContextKey.CURRENT_JOB_ID, "")
+        self.log_error(fl_ctx, f"Job {job_id} is blocked: {reason}")
+        fl_ctx.set_prop(key=FLContextKey.AUTHORIZATION_REASON, value=reason)
+        fl_ctx.set_prop(key=FLContextKey.AUTHORIZATION_RESULT, value=False)
+
+    def _block_job(self, reason: str, fl_ctx: FLContext):
+        job_id = fl_ctx.get_prop(FLContextKey.CURRENT_JOB_ID, "")
+        self.log_error(fl_ctx, f"Job {job_id} is blocked: {reason}")
+        fl_ctx.set_prop(key=FLContextKey.JOB_BLOCK_REASON, value=reason)
```

Comment:
Do you need to add this here?

fl_ctx.set_prop(key=FLContextKey.AUTHORIZATION_RESULT, value=False)

Reply:
Probably we can combine the JOB_BLOCK_REASON check with the AUTHORIZATION_RESULT solution. But we can do that in a future enhancement.